changelog/targetinfo.dd

```diff
@@ -0,0 +1,12 @@
+Add `__traits(getTargetInfo, "key")` and `__traits(allTargetInfos)` to query details about the compilation target.
+
+`getTargetInfo` accepts a string key as argument to select the target detail of interest.
+`allTargetInfos` evaluates to a tuple of strings containing all keys accepted by the implementation.
+
+---
+static assert([ __traits(allTargetInfos) ] == [ "cxxlib" ])
```

Comment:
I can imagine it may be useful to 1. check if a thing you want to query is supported for the target/toolchain, 2. dump as much diagnostic information as available for debug builds and such.

There's a discoverability problem for values that might be supported only on specific backends or toolchains. I think this helps with that.

I added it because someone else asked for it. It was trivial to add, and I can imagine how it might be useful.